pubsub/google/cloud/pubsub/topic.py

```diff
@@ -464,16 +480,28 @@ def publish(self, message, **attrs):
         :param attrs: key-value pairs to send as message attributes
         """
         self.topic._timestamp_message(attrs)
-        self.messages.append(
-            {'data': message,
-             'attributes': attrs})
+
+        # Append the message to the list of messages..
+        item = {'attributes': attrs, 'data': message}
+        self.messages.append(item)
+
+        # Determine the approximate size of the message, and increment
+        # the current batch size appropriately.
+        encoded = b''
+        if isinstance(message, six.text_type):
+            encoded += base64.b64encode(message.encode('utf8'))
+        else:
+            encoded += base64.b64encode(message)
+        encoded += base64.b64encode(
+            json.dumps(attrs, ensure_ascii=False).encode('utf8'),
```

Comment:
The Python docs for `json.dumps` say:

> If `ensure_ascii` is false, all non-ASCII characters are not escaped,
> and  the return value may be a `unicode` instance. See `dump` for details.

And for `json.dump`:

> If ``ensure_ascii`` is true (the default), all non-ASCII characters in the
> output are escaped with ``\uXXXX`` sequences, and the result is a ``str``
> instance consisting of ASCII characters only.  If ``ensure_ascii`` is
> ``False``, some chunks written to ``fp`` may be ``unicode`` instances.
> This usually happens because the input contains unicode strings or the
> ``encoding`` parameter is used. Unless ``fp.write()`` explicitly
> understands ``unicode`` (as in ``codecs.getwriter``) this is likely to
> cause an error.